kit/Kit.cpp

```diff
@@ -2830,30 +2851,57 @@ int KitSocketPoll::kitPoll(int timeoutMicroS)
     }
     else
     {
+        // taskPriority: 100==TaskPriority::DEFAULT, 110==TaskPriority::DEFAULT_IDLE
+        const bool loIdleDefault = (taskPriority > 100 && taskPriority <= 110) ||
+                                   (taskPriority == 100 && timeoutMicroS == 0);
+        const bool loIdleLong = taskPriority > 110 ||
+                                (taskPriority == 100 && timeoutMicroS > 2500000); // LO-IdleHandler's SC_IDLE_MAX = 3s, allowing a 500ms lag
         if (checkForIdle)
             timeoutMicroS = 0;
 
+        auto getTileQueueSize = [&]() -> size_t {
+            return _document ? _document->getTileQueueSize() : 0;
+        };
         // Flush at most maxEvents+1, or return when nothing left.
         _pollEnd = startTime + std::chrono::microseconds(timeoutMicroS);
+        int slice = 0;
```

Comment:
so, if we want to poll until a given timeout, and we know from core that it is "idle" then we split the timeout to do a quick initial poll to see if there are any events, and if there are none we can drain the queue of pending non-priority rendering requests, right?

Does that mean there is no point in realTimeout = std::min(100000, timeoutMicroS), I don't get what difference it makes, but I'm probably missing something. Maybe some comments in here as to the intention would help.

Reply:
added note to comment